libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/MapboxRouteOptionsUpdater.kt

```diff
@@ -127,6 +127,16 @@ class MapboxRouteOptionsUpdater : RouteOptionsUpdater {
                             coordinatesList.size - remainingWaypoints - 1
                         )
                     )
+                    .layersList(
+                        mutableListOf(routeProgress.zLevel).apply {
+                            val legacyLayerList = routeOptions.layersList()
+                            if (legacyLayerList != null) {
+                                addAll(legacyLayerList.takeLast(remainingWaypoints))
+                            } else {
+                                repeat(remainingWaypoints) { add(null) }
+                            }
```

Comment:
The docs say:
> If provided, the list of layers must be the same length as the list of coordinates.

With the proposed impl, the provided list would be of size `coordinates - 1`. We need to fetch the current `zLevel` from the `RouteProgress`/`TripSession` and add it as the first element in this list.

Reply:
Ah you did that above already, never mind!